### PR TITLE
feat: [SRM-10423]: Disable input set in details drawer

### PR DESCRIPTION
**Pull request by deepesh-ui**
##### Summary:

1. No runtime for Monitored ServiceTemplateInputs 
2. Add sources as empty while creating empty templates
3. Read Only for Env and Service 

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by deepesh-ui** [comment]
trigger cypress

**Comment by deepesh-ui** [comment]
trigger cypress

File: src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.utils.ts [hunk 1]
```diff
@@ -31,7 +31,9 @@ export const getInitFormData = (
       type: MonitoredServiceType.APPLICATION as MonitoredServiceForm['type'],
       environmentRef: templateValue?.spec?.environmentRef,
       environmentRefList: [],
-      sources: templateValue?.spec?.sources,
+      sources: isEdit
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.utils.ts:34]: Shouldn't the order of spreading values opposite ?.

...templateValue?.spec?.sources,
sources : isEdit ? .....

File: src/modules/85-cv/pages/monitored-service/MonitoredServiceInputSetsTemplate.tsx [hunk 2]
```diff
@@ -235,11 +235,13 @@ export default function MonitoredServiceInputSetsTemplate({ templateData }: { te
                     name={'serviceRef'}
                     label={getString('cv.healthSource.serviceLabel')}
                     selectItems={serviceOptions}
+                    disabled={isTemplateDrawer}
                   />
                   <FormInput.MultiTypeInput
                     name={'environmentRef'}
                     label={getString('cv.healthSource.environmentLabel')}
                     selectItems={environmentOptions}
+                    disabled={isTemplateDrawer}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/MonitoredServiceInputSetsTemplate.tsx:244]: What does the variable isTemplateDrawer signifies ? Can we some intutive variable name here ?

Comment (praneshg239) [reply src/modules/85-cv/pages/monitored-service/MonitoredServiceInputSetsTemplate.tsx:244]: +1

File: src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.utils.ts (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.utils.ts:34]: No , we need to populate value of change source  only while creating the monitored service template

File: src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx [hunk 3]
```diff
@@ -60,8 +60,8 @@ export default function Configurations(
   const history = useHistory()
   const { isTemplate } = useMonitoredServiceContext()
   const { expressions } = useVariablesExpression()
-  const { orgIdentifier, projectIdentifier, accountId, identifier } = useParams<
-    ProjectPathProps & { identifier: string }
+  const { orgIdentifier, projectIdentifier, accountId, identifier, templateIdentifier } = useParams<
+    ProjectPathProps & { identifier: string } & { templateIdentifier?: string }
```
Comment (praneshg239) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/Configurations.tsx:65]:
```ts
function a<T>(b: T): T{
    return b
}
a<{c: number, d: number}>({c:2, d:3})
```

We can use commas to separate properties in Generic types. This line can be written as 

```ts
useParams<ProjectPathProps & { identifier: string, templateIdentifier?: string}>()
```

**Review by harshilbits — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**